src/gsy_e/models/strategy/load_hours.py

```diff
@@ -491,37 +553,18 @@ def _calculate_active_markets(self):
         ] if self.area else []
 
     def _is_market_active(self, market):
-        return (self._allowed_operating_hours(market.time_slot) and market.in_sim_duration and
+        return (self._energy_params._allowed_operating_hours(market.time_slot) and market.in_sim_duration and
                 (not self.area.current_market or
                  market.time_slot >= self.area.current_market.time_slot))
 
-    def assign_hours_of_per_day(self, hrs_of_day: List[int], hrs_per_day: int):
-        """Validate and assign the values of hrs_of_day and _initial_hrs_per_day."""
-        if hrs_of_day is None:
-            hrs_of_day = list(range(24))
-
-        # be a parameter on the constructor or if we want to deal in percentages
-        if hrs_per_day is None:
-            hrs_per_day = len(hrs_of_day)
-
-        self.hrs_of_day = hrs_of_day
-        self._initial_hrs_per_day = hrs_per_day
-
-        if not all(0 <= h <= 23 for h in hrs_of_day):
-            raise ValueError("Hrs_of_day list should contain integers between 0 and 23.")
-
-        if len(hrs_of_day) < hrs_per_day:
-            raise ValueError("Length of list 'hrs_of_day' must be greater equal 'hrs_per_day'")
-
     def _update_energy_requirement_future_markets(self):
         if not GlobalConfig.FUTURE_MARKET_DURATION_HOURS:
             return
         for time_slot in self.area.future_market_time_slots:
-            if self._allowed_operating_hours(time_slot):
+            if self._energy_params._allowed_operating_hours(time_slot):
```

Comment:
Nit, It seems we're reaching out to a private member of the `_energy_params` object, do you think making it public makes sense?

Reply:
Hmm, I think that you are reviewing it per-commit that's why you do not see it yet. I have converted this to public on a later commit.